buildsystem/modules/FindPython.cmake

```diff
@@ -170,19 +170,24 @@ endforeach()
 # test all the found interpreters; break on success.
 foreach(PYTHON ${PYTHON_INTERPRETERS})
 
-	# ask the interpreter for the essential extension-building flags
-	py_get_config_var(INCLUDEPY PYTHON_INCLUDE_DIR)
-	py_get_config_var(LIBDIR PYTHON_LIBRARY_DIR)
-	py_get_lib_name(PYTHON_LIBRARY_NAME)
+	if (PYTHON STREQUAL PYTHON_EXECUTABLE)
+		find_package(PythonLibs REQUIRED)
+	else()
+		# ask the interpreter for the essential extension-building flags
+		py_get_config_var(INCLUDEPY PYTHON_INCLUDE_DIRS)
+		py_get_config_var(LIBDIR PYTHON_LIBRARY_DIR)
+		py_get_lib_name(PYTHON_LIBRARY_NAME)
+		set(PYTHON_LIBRARIES "${PYTHON_LIBRARY_DIR}/${PYTHON_LIBRARY_NAME}")
+	endif()
 
 	# there's a static_assert that tests the Python version.
 	# that way, we verify the interpreter and the library version.
 	# (the interpreter provided us the library location)
 	try_compile(PYTHON_TEST_RESULT
 		"${CMAKE_BINARY_DIR}"
 		SOURCES "${CMAKE_CURRENT_LIST_DIR}/FindPython_test.cpp"
-		LINK_LIBRARIES "${PYTHON_LIBRARY_NAME}"
-		CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${PYTHON_INCLUDE_DIR}" "-DLINK_DIRECTORIES=${PYTHON_LIBRARY_DIR}" "-DCMAKE_CXX_STANDARD=14"
+		LINK_LIBRARIES ${PYTHON_LIBRARIES}
+		CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${PYTHON_INCLUDE_DIRS}" "-DCMAKE_CXX_STANDARD=14"
```

Comment:
The variable `PYTHON_LIBRARIES` has the full path of the libraries (actually just one library per configuration) that need to be linked. I made (rather, tried to make) the variable consistent for other ways to locate the Python interpreter and libraries. I must've missed something that made kevin fail.